public/scripts/openai.js

```diff
@@ -3541,8 +3591,23 @@ async function onExportPresetClick() {
 
     const preset = structuredClone(openai_settings[openai_setting_names[oai_settings.preset_settings_openai]]);
 
-    delete preset.reverse_proxy;
-    delete preset.proxy_password;
+    const fieldValues = sensitiveFields.filter(field => preset[field]).map(field => `<b>${field}</b>: <code>${preset[field]}</code>`);
+    const shouldConfirm = fieldValues.length > 0;
+    const textHeader = 'Your preset contains proxy and/or custom endpoint settings.';
+    const textMessage = `<div>Do you want to remove these fields before exporting?</div><br>${DOMPurify.sanitize(fieldValues.join('<br>'))}`;
+    const cancelButton = { text: 'Cancel', result: POPUP_RESULT.CANCELLED, appendAtEnd: true, action: () => popup.complete(POPUP_RESULT.CANCELLED) };
+    const popupOptions = { customButtons: [cancelButton] };
+    const popup = new Popup(PopupUtils.BuildTextWithHeader(textHeader, textMessage), POPUP_TYPE.CONFIRM, '', popupOptions);
```

Comment:
Do I need to understand this? Why don't you just use `Popup.show.confirm`, @Cohee1207? You can still await the result below, if you don't want to do it in one line.
And yes, that allow popup options to be passed into too.

Reply:
Because I need a popup instance in button callback.

Reply:
Resolved with recent staging changes